src/core/memory.cpp

```diff
@@ -64,9 +64,21 @@ PAddr MemoryManager::Allocate(PAddr search_start, PAddr search_end, size_t size,
 void MemoryManager::Free(PAddr phys_addr, size_t size) {
     std::scoped_lock lk{mutex};
 
-    const auto dmem_area = FindDmemArea(phys_addr);
-    ASSERT(dmem_area != dmem_map.end() && dmem_area->second.base == phys_addr &&
-           dmem_area->second.size == size);
+    auto dmem_area = FindDmemArea(phys_addr);
+    ASSERT(dmem_area != dmem_map.end() && dmem_area->second.size >= size);
```

Comment:
Does P.T do partial direct memory area free as well (I tested it on linux and it didn't hit this assert)? Unless it does that better to leave this unchanged

Reply:
Yes, the version I tested allocates 0-7f0000, then frees 0-300000, followed by 700000-7f0000.

Reply:
In that case I would recommend doing similar thing to VMA logic. Change the assertion to a Contains() check and use CarveDmemArea at the bottom of the function